### PR TITLE
login

### PR DESCRIPTION
**Pull request by Erikaguilera123**
## Sumary

Code of login to a user registed in the page

### CODE REVIEW

**Comment by Erikaguilera123** [comment]
El email and password not are in the .env because is the a test page

File: cypress/const/constants.js [hunk 1]
```diff
@@ -8,7 +8,10 @@ export const VAR = {
   nulo:' ',
   emailInvalid: 'email is invalid',
   passInvalid:'password is too short (minimum is 8 characters)',
-  userInvalid:"username can't be blank"
+  userInvalid:"username can't be blank",
```
Comment (betoobernal) [inline cypress/const/constants.js:11]: use single quotes

File: cypress/integration/loginAngular.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import { VAR, URL } from '../const/constants';
+import { IDs } from '../page-objects/loginAngular';
+import{ HAVETEXT } from '../const/helpers';
```
Comment (betoobernal) [inline cypress/integration/loginAngular.spec.js:3]:
```suggestion
import { HAVETEXT } from '../const/helpers';
```
add a space after `import`

Comment (betoobernal) [inline cypress/integration/loginAngular.spec.js:3]: I think that this helper should be in snake_case 
```suggestion
import{ HAVE_TEXT } from '../const/helpers';
```

File: cypress/page-objects/loginAngular.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
+import { VAR } from '../const/constants';
+
+export const IDs = {
+  idSingIn: 'a[href*="/login"]',
```
Comment (betoobernal) [inline cypress/page-objects/loginAngular.js:5]: wrong typing
```suggestion
  idSignIn: 'a[href*="/login"]',
```

File: cypress/const/constants.js (hunk 1 above)
Comment (eledwinn) [reply cypress/const/constants.js:11]: And leave a white space after `:`

File: cypress/page-objects/loginAngular.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  email: 'input[formcontrolname*="email"]',
+  password: 'input[formcontrolname*="password"]',
+  btnSingIn: 'button[type*="submit"]',
```
Comment (betoobernal) [inline cypress/page-objects/loginAngular.js:8]: wrong typing
```suggestion
  btnSingIn: 'button[type*="submit"]',
```

File: cypress/page-objects/loginAngular.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  errorRegister: 'error-messages',
+  nameProfile: 'a[href*="/profile/prueba%20name"]',
```
Comment (betoobernal) [inline cypress/page-objects/loginAngular.js:10]:
```suggestion
  profileName: 'a[href*="/profile/prueba%20name"]',
```

File: cypress/page-objects/loginAngular.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+
+import { VAR } from '../const/constants';
+
+export const IDs = {
+  idSingIn: 'a[href*="/login"]',
+  email: 'input[formcontrolname*="email"]',
+  password: 'input[formcontrolname*="password"]',
+  btnSingIn: 'button[type*="submit"]',
+  errorRegister: 'error-messages',
```
Comment (betoobernal) [inline cypress/page-objects/loginAngular.js:9]:
```suggestion
  registerError: 'error-messages',
```

File: cypress/integration/loginAngular.spec.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+   });
+
+  it('Sign up data incorrect', () => { 
+    
+    HAVETEXT(IDs.email,VAR.textSmall);
+    HAVETEXT(IDs.password,VAR.textSmall);
+    cy.get(IDs.btnSingIn).click();
+
+    cy.get(`.${IDs.errorRegister}`).contains(VAR.error);
+
```
Comment (eledwinn) [inline cypress/integration/loginAngular.spec.js:23]: Please remove empty lines in start and end of `it(` block

File: cypress/page-objects/registroAngular.js [hunk 8]
```diff
@@ -2,11 +2,10 @@
 import { VAR } from '../const/constants';
 
 export const IDs = {
-  idSingIn: 'a[href*="#login"]',
   idSingUp:  'a[href*="/register"]',
```
Comment (betoobernal) [inline cypress/page-objects/registroAngular.js:5]: wrong tying 
```suggestion
  idSignUp:  'a[href*="/register"]',
```

File: cypress/integration/loginAngular.spec.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+   });
+
+  it('Sign up data incorrect', () => { 
+    
+    HAVETEXT(IDs.email,VAR.textSmall);
+    HAVETEXT(IDs.password,VAR.textSmall);
+    cy.get(IDs.btnSingIn).click();
+
+    cy.get(`.${IDs.errorRegister}`).contains(VAR.error);
```
Comment (eledwinn) [inline cypress/integration/loginAngular.spec.js:22]: If we use constants to save the selectors, we should save the entire selector.
Example: the `IDs.errorRegister` shoud have `.error-messages`, not only `error-messages`.

File: cypress/integration/loginAngular.spec.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+  });
+
+  it('Sign in', () => { 
+   HAVETEXT(IDs.email,VAR.emailRegister);
+   HAVETEXT(IDs.password,VAR.passRegister);
+
+   cy.get(IDs.btnSingIn).click();
+   cy.get(IDs.nameProfile).should('be.visible');
+  });
+
```
Comment (eledwinn) [inline cypress/integration/loginAngular.spec.js:33]: Remove empty line please 😃

File: cypress/page-objects/loginAngular.js [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+  nameProfile: 'a[href*="/profile/prueba%20name"]',
+
```
Comment (eledwinn) [inline cypress/page-objects/loginAngular.js:11]: Same

File: cypress/page-objects/loginAngular.js [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+import { VAR } from '../const/constants';
+
+export const IDs = {
```
Comment (eledwinn) [inline cypress/page-objects/loginAngular.js:4]: Not all here are IDs, but no use another name like `selectors`?

File: cypress/page-objects/loginAngular.js (hunk 5 above)
Comment (eledwinn) [inline cypress/page-objects/loginAngular.js:10]: Could we find another selector? this could change very soon. :/

**Review by eledwinn — CHANGES_REQUESTED** [review]

File: cypress/const/constants.js (hunk 1 above)
Comment (Erikaguilera123) [reply cypress/const/constants.js:11]: Change in the branch "NewArticule"

File: cypress/integration/loginAngular.spec.js (hunk 2 above)
Comment (Erikaguilera123) [reply cypress/integration/loginAngular.spec.js:3]: Change in the branch "NewArticule"

Comment (Erikaguilera123) [reply cypress/integration/loginAngular.spec.js:3]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 3 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:5]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 4 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:8]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 5 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:10]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 6 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:9]: Change in the branch "NewArticule"

File: cypress/page-objects/registroAngular.js (hunk 8 above)
Comment (Erikaguilera123) [reply cypress/page-objects/registroAngular.js:5]: Change in the branch "NewArticule"

File: cypress/integration/loginAngular.spec.js (hunk 7 above)
Comment (Erikaguilera123) [reply cypress/integration/loginAngular.spec.js:23]: Change in the branch "NewArticule"

File: cypress/integration/loginAngular.spec.js (hunk 9 above)
Comment (Erikaguilera123) [reply cypress/integration/loginAngular.spec.js:22]: Change in the branch "NewArticule"

File: cypress/integration/loginAngular.spec.js (hunk 10 above)
Comment (Erikaguilera123) [reply cypress/integration/loginAngular.spec.js:33]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 11 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:11]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 5 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:10]: Change in the branch "NewArticule"

File: cypress/page-objects/loginAngular.js (hunk 12 above)
Comment (Erikaguilera123) [reply cypress/page-objects/loginAngular.js:4]: Change in the branch "NewArticule"

**Comment by Erikaguilera123** [comment]
Change in the branch "NewArticule"

**Pull request closed without merging**